### ISSUE TITLE
CMake does not generate "toFloat.h" with Ninja

### ISSUE THREAD

**Issue by ghost** [openexr/openexr#94]
On Windows. Can someone check that?

```
ninja: error: 'C:/Users/Constantine/Desktop/openexr/IlmBase/Half/toFloat.h', needed
by 'Half/CMakeFiles/Half.dir/half.cpp.obj', missing and no known rule to make it
```

**Comment by francis-brissette** [comment]
I have the same issue with Ninja on Window:
`
cd openexr\IlmBase
mkdir build
cd build
cmake .. -G Ninja
cmake --build .
`

Output:
`
ninja: error: 'Half/toFloat.h', needed by 'Half/CMakeFiles/Half.dir/half.cpp.obj', missing and no known rule to make it
`

Anyone have a fix?

**Comment by ZeroCrunch** [comment]
Anyone managed to get this working?

**Comment by francis-brissette** [comment]
@ZeroCrunch ,

In openexr/IlmBase/Half/CMakeLists.txt, I updated the code to the followin:g:

...
SET_SOURCE_FILES_PROPERTIES(
  half.cpp
  PROPERTIES
  OBJECT_DEPENDS
  toFloat
  OBJECT_DEPENDS
  eLut
  )

IF(BUILD_SHARED_LIBS)
    ADD_DEFINITIONS(-DHALF_EXPORTS)
ENDIF()

ADD_LIBRARY ( Half ${LIB_TYPE}
  half.cpp
  )

INSTALL ( TARGETS
  Half
  DESTINATION
  lib
)
...

It worked on my side.

Cheers,
Francis

**Comment by meshula** [comment]
This looks like the correct fix to me - it would be a good PR.

**Comment by ZeroCrunch** [comment]
@francis-brissette,

What worked for me in the end was:

PROPERTIES
      OBJECT_DEPENDS
      "${CMAKE_CURRENT_BINARY_DIR}/toFloat.h;${CMAKE_CURRENT_BINARY_DIR}/eLut.h"

**Comment by francis-brissette** [comment]
@meshula, will you take care of this bug & fix?

**Comment by meshula** [comment]
@ZeroCrunch @francis-brissette This fix is on the develop branch, thanks Jamie!

**Comment by cary-ilm** [comment]
Looking into the backlog of open OpenEXR issues. This seems to have been resolved in the v2.3.0 release.
Closing the issue for now, feel free to re-open or file a new issue if you need further help.
